sdk/src/azure/core/az_json_token.c

```diff
@@ -339,19 +336,28 @@ AZ_NODISCARD az_result az_json_string_unescape(
   _az_PRECONDITION(destination_max_size > 0);
 
   int32_t position = 0;
-  for (int32_t i = 0; i < json_string._internal.size; i++)
+  int32_t span_size = az_span_size(json_string);
+  uint8_t* span_ptr = az_span_ptr(json_string);
+  for (int32_t i = 0; i < span_size; i++)
   {
-    uint8_t current_char;
-    current_char = json_string._internal.ptr[i];
-    if (current_char == '\\' && i < json_string._internal.size)
+    uint8_t current_char = span_ptr[i];
+    if (current_char == '\\' && i < span_size - 1)
     {
-      uint8_t next_char = json_string._internal.ptr[i + 1];
-      // check that we have sometig to escape
+      uint8_t next_char = span_ptr[i + 1];
+      // check that we have something to escape
       if (_az_is_valid_escaped_character(next_char))
       {
         current_char = _az_json_unescape_single_byte(next_char);
         i++;
       }
+      else
+      {
+        return AZ_ERROR_UNEXPECTED_CHAR;
+      }
+    }
+    else if (current_char == '\\' && i == span_size - 1)
```

Comment:
Test coverage is showing this branch isn't fully covered (likely one of the 3 cases where the condition will be false is missing).

Reply:
i would say the code covreage tool is dumb. the branch not covered here , setting breakpoints on all the branches , get hit, there is one branch which is not explicitly written in the code but it is executed, the else on the last else if(). Since the condition returns the else branch continues the execution. i would rather not specifically put the else branch as it would make the code look blah.

Reply:
Hmm, I think I know why. Parts of the condition are redundant, so they can never be covered. It's not because of a missing else.

You can remove `&& i == span_size - 1`, and the code will be identical.

The way the code is written, I don't think there's any way for `current_char == '\\' ` to be true, in this else-if but `i == span_size - 1` to be false.

The for-loop puts a bound on i to be between 0 and span_size.
If i is not less than span_size - 1 (from the if-check), then it must be equal to it.

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-c/pull/2291